code/ai/ailua.cpp

```diff
@@ -67,8 +214,13 @@ void run_ai_lua_action(const luacpp::LuaFunction& action, const ai_mode_lua& lua
 	auto retVals = action.call(Script_system.GetLuaSession(), luaParameters);
 
 	if (!retVals.empty() && retVals[0].getValueType() == luacpp::ValueType::BOOLEAN) {
-		if (retVals[0].getValue<bool>())
-			ai_mission_goal_complete(aip);
+		if (retVals[0].getValue<bool>()) {
+
+			// If we don't have a ship then it's a general order and we have nothing else to do
+			if (aip->shipnum < 0) {
```

Comment:
Isn't this the wrong way?
We only need to complete the goal, if the shipnum is larger or equal to 0 and valid?

Reply:
yes, I believe you are correct